modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/PrecisionAtK.java

```diff
@@ -65,42 +70,52 @@
      *            Set to 'true', unlabeled documents are ignored and neither count
      *            as true or false positives. Set to 'false', they are treated as
      *            false positives.
+     * @param k
+     *            controls the window size for the search results the metric takes into account
      */
-    public PrecisionAtK(int threshold, boolean ignoreUnlabeled) {
+    public PrecisionAtK(int threshold, boolean ignoreUnlabeled, int k) {
         if (threshold < 0) {
             throw new IllegalArgumentException("Relevant rating threshold for precision must be positive integer.");
         }
+        if (k <= 0) {
+            throw new IllegalArgumentException("Window size k must be positive.");
+        }
         this.relevantRatingThreshhold = threshold;
         this.ignoreUnlabeled = ignoreUnlabeled;
+        this.k = k;
     }
 
     public PrecisionAtK() {
-        this(1, false);
+        this(1, false, DEFAULT_K);
     }
 
-
     private static final ConstructingObjectParser<PrecisionAtK, Void> PARSER = new ConstructingObjectParser<>(NAME,
             args -> {
                 Integer threshHold = (Integer) args[0];
                 Boolean ignoreUnlabeled = (Boolean) args[1];
+                Integer k = (Integer) args[2];
                 return new PrecisionAtK(threshHold == null ? 1 : threshHold,
-                        ignoreUnlabeled == null ? false : ignoreUnlabeled);
+                        ignoreUnlabeled == null ? false : ignoreUnlabeled,
+                                k == null ? 10 : k);
```

Comment:
Do we want to put here `DEFAULT_K` instead of `10`?

Reply:
Yes, also already changed that on my local branch.